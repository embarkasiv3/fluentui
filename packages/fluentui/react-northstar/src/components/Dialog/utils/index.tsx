const _disableIosBodyScroll = (event: TouchEvent) => {
  event.preventDefault();
};

export function enableBodyScroll(target: HTMLElement): void {
  if (target) {
    target.style.overflow = 'unset';
    target.removeEventListener('touchmove', _disableIosBodyScroll);
  }
}

export function disableBodyScroll(target: HTMLElement): void {
  if (target) {
    target.style.overflow = 'hidden';
    target.addEventListener('touchmove', _disableIosBodyScroll, { passive: false, capture: false });
  }
}

const dialogsCounterAttribute = 'fluent-dialogs-count';

export const lockBodyScroll = (target: Document) => {
  const openDialogs = (+target.body.getAttribute(dialogsCounterAttribute) || 0) + 1;
  target.body.setAttribute(dialogsCounterAttribute, `${openDialogs}`);

  // Avoid to block scroll in nested dialogs
  if (openDialogs === 1) {
    disableBodyScroll(target.body);
  }
};

export const unlockBodyScroll = (target: Document) => {
  const openDialogs = (+target.body.getAttribute(dialogsCounterAttribute) || 0) - 1;
  target.body.setAttribute(dialogsCounterAttribute, `${openDialogs}`);

  // Only enables scroll if all dialogs are closed
  if (openDialogs === 0) {
    enableBodyScroll(target.body);
    target.body.removeAttribute(dialogsCounterAttribute);
  }
};
