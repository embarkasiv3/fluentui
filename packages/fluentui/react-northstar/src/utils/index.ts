export { childrenExist } from './childrenExist';
export { rtlTextContainer } from './rtlTextContainer';
export { stringLiteralsArray } from './stringLiteralsArray';
export { getOrGenerateIdFromShorthand } from './getOrGenerateIdFromShorthand';

export * from './factories';
export { mergeProviderContexts } from './mergeProviderContexts';

export * from './renderComponent';
export { renderComponent } from './renderComponent';

export { htmlImageProps, htmlInputAttrs, htmlInputEvents, htmlInputProps, partitionHTMLProps } from './htmlPropsUtils';

export { isBrowser } from './isBrowser';
export { doesNodeContainClick } from './doesNodeContainClick';

export { pxToRem } from './fontSizeUtility';
export { createComponent } from './createComponent';
export { getKindProp } from './getKindProp';
export * from './whatInput';

export * from './commonPropInterfaces';
export { screenReaderContainerStyles } from './accessibility/Styles/accessibilityStyles';
// work around api-extractor limitation
import { CreateCommonConfig as CreateCommonConfigLocal, createCommon as createCommonLocal } from './commonPropTypes';

export module commonPropTypes {
  export type CreateCommonConfig = CreateCommonConfigLocal;
  export const createCommon = createCommonLocal;
}
