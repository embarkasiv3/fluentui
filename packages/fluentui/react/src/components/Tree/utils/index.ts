import * as _ from 'lodash';
import { TreeItemProps } from '../TreeItem';
import { ShorthandValue } from '../../../types';

export const hasSubtree = (item: TreeItemProps | ShorthandValue<TreeItemProps>): boolean => {
  return !_.isNil(item['items']) && item['items'].length > 0;
};

export const removeItemAtIndex = (items: any[], itemIndex: number): any[] => {
  return [...items.slice(0, itemIndex), ...items.slice(itemIndex + 1)];
};
