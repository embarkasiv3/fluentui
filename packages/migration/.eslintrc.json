{
  "extends": ["plugin:@fluentui/eslint-plugin/node--legacy"],
  "root": true,
  "overrides": [
    {
      // These files are not actually compiled and therefore don't need correct deps and such
      // (they just have to be syntactically valid)
      "files": ["fixtures/**/*.{ts,tsx}"],
      "rules": {
        "import/no-extraneous-dependencies": "off"
      }
    }
  ]
}
