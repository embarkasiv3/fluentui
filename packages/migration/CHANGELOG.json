{
  "name": "@uifabric/migration",
  "entries": [
    {
      "version": "0.1.2",
      "tag": "@uifabric/migration_v0.1.2",
      "date": "Wed, 20 Mar 2019 03:15:21 GMT",
      "comments": {
        "patch": [
          {
            "comment": "Migration: fixed the bin script's shebang to point to the right \"env\" executable",
            "author": "Kenneth Chau <kenotron@users.noreply.github.com>",
            "commit": "693183132f82cf336b4e8f8fa00f56050882dcfc"
          }
        ]
      }
    },
    {
      "version": "0.1.1",
      "tag": "@uifabric/migration_v0.1.1",
      "date": "Tue, 19 Mar 2019 18:05:11 GMT",
      "comments": {
        "patch": [
          {
            "comment": "Initial release, under construction",
            "author": "Kenneth Chau <kenotron@users.noreply.github.com>",
            "commit": "7a3bb677a144c1be006c7e7c7afe8ed9b8c353ae"
          }
        ]
      }
    }
  ]
}
