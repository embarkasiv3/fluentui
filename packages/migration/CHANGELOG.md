# Change Log - @uifabric/migration

This log was last generated on Thu, 21 Mar 2019 17:21:42 GMT and should not be manually modified.

## 0.1.3
Thu, 21 Mar 2019 17:21:42 GMT

### Patches

- migration: the bin directory was being ignored, this change makes sure the npm publish step will include the bin scripts

## 0.1.2
Wed, 20 Mar 2019 03:15:21 GMT

### Patches

- Migration: fixed the bin script's shebang to point to the right "env" executable

## 0.1.1
Tue, 19 Mar 2019 18:05:11 GMT

### Patches

- Initial release, under construction

