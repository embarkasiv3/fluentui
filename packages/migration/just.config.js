const { preset, just } = require('@uifabric/build');
const { task, series } = just;

preset();
// TODO remove, fix tests on windows environments
// TODO: add "test" command to migration/package.json once this merges: https://github.com/OfficeDev/office-ui-fabric-react/pull/11614
task('build', series('clean', 'copy', 'ts:commonjs-only')).cached();
