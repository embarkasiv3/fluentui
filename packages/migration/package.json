{
  "name": "@uifabric/migration",
  "version": "0.2.2",
  "description": "Migration scripts for UI Fabric",
  "main": "lib/index.js",
  "bin": {
    "migration": "./bin/migration.js"
  },
  "typings": "lib/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/OfficeDev/office-ui-fabric-react"
  },
  "license": "MIT",
  "scripts": {
    "build": "node ../../scripts/just.js build-commonjs-only",
    "clean": "node ../../scripts/just.js clean",
    "code-style": "node ../../scripts/just.js code-style",
    "start": "node ../../scripts/just.js dev",
    "start-test": "node ../../scripts/just.js jest-watch",
    "update-snapshots": "node ../../scripts/just.js jest -u"
  },
  "devDependencies": {
    "@types/jest": "23.0.0",
    "@types/glob": "7.1.1",
    "@types/mock-fs": "3.6.30",
    "@types/node": "^8.10.29",
    "@types/semver": "^5.5.0",
    "@uifabric/prettier-rules": "^1.0.2",
    "@uifabric/tslint-rules": "^1.0.2",
    "glob": "^7.1.3",
    "mock-fs": "4.10.0"
  },
  "dependencies": {
    "semver": "^5.6.0",
    "tslib": "^1.7.1",
    "riceburn": "^1.3.0",
    "typescript": "3.3.3"
  },
  "disabledTasks": [
    "verify-api-extractor"
  ]
}
