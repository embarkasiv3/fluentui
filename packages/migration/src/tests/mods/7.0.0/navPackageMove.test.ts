import migration from '../../../mods/7.0.0/navToLegacy';
import path from 'path';
import { runMigration } from '../utils';

describe('Nav package move', () => {
  it('moves import statements', async () => {
    const result = await runMigration(migration, path.join('7.0.0', 'nav.tsx'));
    expect(result.warnings).toEqual([]);
    expect(result.contents).toMatchSnapshot();
  });
});
