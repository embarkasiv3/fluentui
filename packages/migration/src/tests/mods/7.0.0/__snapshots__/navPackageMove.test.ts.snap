// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav package move moves import statements 1`] = `
"// prettier-ignore
import { Nav, INavLink } from '@uifabric/legacy';
// prettier-ignore
import { Nav } from '@uifabric/legacy';
// prettier-ignore
// tslint:disable-next-line: quotemark
import { Nav } from '@uifabric/legacy';
// prettier-ignore
import { INav, INavLink, INavLinkGroup, INavProps, INavState, INavStyleProps, INavStyles, isRelativeUrl, Nav, NavBase } from '@uifabric/legacy';
// prettier-ignore
import { Foo, Bar } from 'office-ui-fabric-react';
import { INav, INavLink, INavLinkGroup, INavProps, INavState, INavStyleProps, INavStyles, isRelativeUrl, Nav, NavBase } from '@uifabric/legacy';
// prettier-ignore
import {
  INav,
  INavLink,
  INavLinkGroup,
  INavProps,
  INavState,
  INavStyleProps,
  INavStyles,
  isRelativeUrl,
  Nav,
  NavBase
} from 'somewhereelse';
"
`;
